sgx-types/src/ssa.rs

```diff
@@ -315,18 +364,13 @@ impl Footer {
             _ => None,
         }
     }
-}
 
-/// When an AEX occurs while running in an enclave, the architectural state is saved
-/// in the thread’s current StateSaveArea (SSA Frame), which is pointed to by TCS.CSSA.
-///
-/// Section 38.9, Table 38-7
-#[repr(C)]
-pub struct StateSaveArea {
-    ///  Section 38.9, Table 38-7
-    pub xsave: XSave,
-    /// Section 38.9, Table 38-7
-    pub footer: Footer,
+    const fn padding() -> usize {
+        #[repr(C, align(4096))]
+        struct Unpadded(XSave, Miscellaneous, Gpr);
+
+        size_of::<Unpadded>() - size_of::<XSave>() - size_of::<Miscellaneous>() - size_of::<Gpr>()
```

Comment:
I just realized that the following is probably more readable.

```rust
        #[repr(C, align(4096))]
        struct Aligned(XSave, Miscellaneous, Gpr);

        #[repr(C, packed)]
        struct Unaligned(XSave, Miscellaneous, Gpr);

        size_of::<Aligned>() - size_of::<Unaligned>()
```

Reply:
If `XSave` is 64-byte aligned (which I'm not sure; see comment above) then this won't work because `error[E0588]: packed type cannot transitively contain a `[repr(align)]` type`.

Reply:
Okay then. Never mind. :)